### ISSUE TITLE
p2p: seed_mode sometimes fails to send pexAddrsMessage before disconnecting gracefully

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#2092]
When `SeedMode` is true, the PexReactor sends a batch of addresses and calls StopPeerGracefully. It seems sometimes the peer ends up disconnecting before these messages are delivered, which makes it difficult for a new node to join the network.

It looks like we don't call .flush() in the MConnection.OnStop(). I tried adding that on the seed node, but even still the new node sometimes fails to receive any addresses from the seed.

**Comment by phymbert** [comment]
@ebuchman `StopPeerGracefully` [yet implemented](https://github.com/tendermint/tendermint/blob/master/p2p/switch.go#L294). `bcStatusResponseMessage` is processed before `pexAddrsMessage`. Connection is closed by seed peer before `pexAddrsMessage` arrived in pex channel...

I did a dirty fix that seem to work in [pex_reactor.go](https://github.com/tendermint/tendermint/blob/master/p2p/pex/pex_reactor.go#L222):
```go
if r.config.SeedMode {
			r.SendAddrs(src, r.book.GetSelectionWithBias(biasToSelectNewPeers))
			go func() {
				time.Sleep(5 * time.Second)
				r.Switch.StopPeerGracefully(src)
			}()
		}
```

I will try to submit a PR to handle shutdown gracefully properly.

**Comment by ebuchman** [comment]
@phymbert you solution was implemented in #2797. But see #2802 for my attempt to do it more correctly.
